src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -413,7 +432,11 @@ private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byt
             RejectRequestHeader(headerSpan);
         }
 
-        return headerSpan.Length;
+        if (array is not null)
+        {
+            ArrayPool<byte>.Shared.Return(array);
```

Comment:
Shouldn't the code always execute this? `RejectRequestHeader` two lines above can throw.

Reply:
https://github.com/dotnet/aspnetcore/pull/45044#discussion_r1020762763

Reply:
Realized I might not be the only one with this comment, and looked at resolved ones ... thanks

Reply:
Actually, why not return it before calling `RejectRequestHeader`, it does require a try/catch, and it might not be that rare to hit this code path.